adafruit_cursorcontrol/adafruit_cursorcontrol_cursormanager.py

```diff
@@ -0,0 +1,127 @@
+# Copyright (c) 2019 Brent Rubell for Adafruit Industries
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+"""
+`adafruit_cursorcontrol_cursormanager`
+================================================================================
+Simple interaction user interface interaction for Adafruit_CursorControl.
+* Author(s): Brent Rubell
+"""
+import board
+import digitalio
+import analogio
+from micropython import const
+from gamepadshift import GamePadShift
+
+# PyBadge
+PYBADGE_BUTTON_LEFT = const(128)
+PYBADGE_BUTTON_UP = const(64)
+PYBADGE_BUTTON_DOWN = const(32)
+PYBADGE_BUTTON_RIGHT = const(16)
+PYBADGE_BUTTON_A = const(2)
+PYBADGE_BUTTON_B = const(1)
+PYBADGE_BUTTON_SEL = const(8)
+PYBADGE_BUTTON_START = const(4)
+
+class CursorManager:
+    """Simple interaction user interface interaction for Adafruit_CursorControl.
+
+    :param adafruit_cursorcontrol cursor: The cursor object we are using.
+    :param adafruit_button.Button buttons: Optional array of button elements.
+    """
+    def __init__(self, cursor, buttons=None):
+        self._cursor = cursor
+        self._btns = buttons
+        self._is_held = False
+        self._init_hardware()
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exception_type, exception_value, traceback):
+        self.deinit()
+
+    def deinit(self):
+        """Deinitializes a CursorManager object."""
+        self._is_deinited()
+        self._pad.deinit()
+        self._cursor_deinit()
+        self._cursor = None
+
+    def _is_deinited(self):
+        """Checks if CursorManager object has been deinitd."""
+        if self._cursor is None:
+            raise ValueError("CursorManager object has been deinitialized and can no longer "
+                             "be used. Create a new CursorManager object.")
+
+    def _init_hardware(self):
+        """Initializes PyBadge or PyGamer hardware."""
+        if hasattr(board, 'BUTTON_CLOCK') and not hasattr(board, 'SD_CS'):
```

Comment:
i think start with if it has joystick x/y so you aren't testing for SD_CS which is a different pin (and may not exist on some future boards).

Reply:
Switched to joystick